### PR TITLE
Implemented the validation for all the fields on the "Sign up" form #131

### LINKED ISSUE #131: (SP: 1) Implement the validation for all the fields on the form
UserStory: #57 

## Description
Implement the validation for all the fields on the sign-up form. 
## Design 
image

## Acceptance Criteria
- [x] "First Name" and "Last Name" fields must receive a string in a range from 2 to 15 symbols.
- [x] The email field must be for emails.
- [x] Password must contain at least one alphabetic and one numeric character.
- [x] The "Confirm Password" field must be connected with the "Password" field.

### PR DESCRIPTION
**Pull request by Olenka-Hryk**
Implemented the validation for all the fields on the `Sign up` form: [ Close #131 ]

Screenshot 2024-05-07 at 17 34 35

- [x] **First Name** and **Last Name** fields only accept strings ranging from 2 to 15 characters:
Screenshot 2024-05-07 at 17 37 00

- [x] The **Email** field is only for email addresses:
Screenshot 2024-05-07 at 17 37 43

- [x] **Password** field must contain at least one alphabetic and one numeric character:
Screenshot 2024-05-07 at 17 38 50

- [x] The **Confirm Password** field connected with the **Password** field:
Screenshot 2024-05-07 at 17 39 55

_As a result, we receive error messages for entered invalid data:_
Screenshot 2024-05-07 at 17 40 51

Additionally, **implemented the disabling of the `Sign up` button when at least one field contains invalid data and the checkbox `I agree to the Terms...` is not selected**:
- _Before making changes:_
Screenshot 2024-05-07 at 17 42 01

- _After making changes_:

| Invalid Data | Valid Data |
|--------------|--------------|
| Invalid Data | Valid Data |

Additionally, **added unit test for `common` and `login` files of `validations`**.
_Current Tests Coverage:_
Screenshot 2024-05-07 at 22 59 58

### CODE REVIEW

**Review by ohanovdmytro — APPROVED** [review]

**Review by Nazar-Lysak — APPROVED** [review]

**Review by docia — APPROVED** [review]

File: src/containers/guest-home-page/signup-form/SignupForm.jsx [hunk 1]
```diff
@@ -25,6 +25,7 @@ const SignupForm = ({
   const { t } = useTranslation()
   const { privacyPolicy, termOfUse } = guestRoutes
   const [isAgreementChecked, setIsAgreementChecked] = useState(false)
+  const [isSignUpButtonDisabled, setIsSignUpButtonDisabled] = useState(true)
```
Comment (vlad-khrychov) [inline src/containers/guest-home-page/signup-form/SignupForm.jsx]: Did you try to use `const isSignUpButtonDisabled` without state and useEffect?

File: tests/unit/utils/validations/common.spec.js
Comment (vlad-khrychov) [inline tests/unit/utils/validations/common.spec.js:1]: Maybe separate files for each `describe` would be more readable?

**Review by vlad-khrychov — APPROVED** [review]
It is great work. Think about do not creating state and hook for the button disabling but you can merge like now.

File: src/containers/guest-home-page/signup-form/SignupForm.jsx (hunk 1 above)
Comment (Olenka-Hryk) [reply src/containers/guest-home-page/signup-form/SignupForm.jsx]: Thank you for the great advice! I've fixed it.

**Pull request merged**